controllers/utils/acm.go

```diff
@@ -0,0 +1,75 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+
+	viewv1beta1 "github.com/stolostron/multicloud-operators-foundation/pkg/apis/view/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	ctrlClient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const MCVLabelKey = "multicluster.odf.openshift.io/cluster"
+const MCVNameTemplate = "odf-multicluster-mcv-%s"
+
+func GetManagedClusterViewName(clusterName string) string {
+	return fmt.Sprintf(MCVNameTemplate, clusterName)
+}
+
+func CreateOrUpdateManagedClusterView(ctx context.Context, client ctrlClient.Client, resourceToFindName string, resourceToFindNamespace string, resourceToFindType string, clusterName string) (*viewv1beta1.ManagedClusterView, controllerutil.OperationResult, error) {
+	mcv := &viewv1beta1.ManagedClusterView{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      GetManagedClusterViewName(clusterName),
+			Namespace: clusterName,
+		},
+	}
+
+	or, err := controllerutil.CreateOrUpdate(ctx, client, mcv, func() error {
+		mcv.Spec = viewv1beta1.ViewSpec{
+			Scope: viewv1beta1.ViewScope{
+				Name:      resourceToFindName,
+				Namespace: resourceToFindNamespace,
+				Resource:  resourceToFindType,
+			},
+		}
+
+		if mcv.Labels == nil {
+			mcv.Labels = make(map[string]string)
+		}
+
+		mcv.Labels[MCVLabelKey] = clusterName
+
+		return nil
+	})
+
+	if err != nil {
+		return nil, controllerutil.OperationResultNone, err
+	}
+
+	return mcv, or, nil
+}
+
+func GetManagedClusterViewByLabel(client ctrlClient.Client, labelKey, labelValue, namespace string) (*viewv1beta1.ManagedClusterView, error) {
```

Comment:
If I know the `labelValue` then I also know the cluster name as `labelValue == clustername`.
Also, `namespace == clustername` since MCV is created in the clustername namespace.
From that I can figure out the MCV name using `GetManagedClusterViewName(clustername)`.
With these info, I can do a simple GET query. So, when do I need to list by label? Can you point me where we must use this?
I am in favor of adding labels where required, but in this particular case it seems redundant.

Reply:
it is not necessary to use labels, we can get it by the name but labeling helps grouping these resources together and works for scalability purposes.
If i would want to fetch all k8s resources with a particular label, it is just more easily doable by label. I was thinking this could help us with future cleanups ?

Reply:
in case we are omitting label way and relying on name, plz add a comment somewhere on top of `GetManagedClusterViewName` function that this name is used for filtering on UI, so that if we change it for any reason in upcoming releases, we remember to fix it in UI as well...

Reply:
@SanjalKatiyar I think it is fine. Its not a major change, if it breaks in UI due to name change , then it can be fixed fast.

Reply:
adding a small informative comment is harmless, but okay, np :)